runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -327,6 +327,59 @@ void J9::RecognizedCallTransformer::process_java_lang_StringUTF16_toBytes(TR::Tr
    treetop->insertAfter(TR::TreeTop::create(comp(), TR::Node::create(node, TR::treetop, 1, newCallNode)));
    }
 
+void J9::RecognizedCallTransformer::process_jdk_internal_util_ArraysSupport_vectorizedMismatch(TR::TreeTop* treetop, TR::Node* node)
+   {
+   TR::Node* a = node->getChild(0);
+   TR::Node* aOffset = node->getChild(1);
+   TR::Node* b = node->getChild(2);
+   TR::Node* bOffset = node->getChild(3);
+   TR::Node* length = node->getChild(4);
+   TR::Node* log2ArrayIndexScale = node->getChild(5);
+   TR::Node* log2ArrayIndexScale64Bits = TR::Node::create(node, TR::iu2l, 1, log2ArrayIndexScale);
+
+   TR::Node* lengthInBytes = TR::Node::create(node, TR::lshl, 2,
+      TR::Node::create(node, TR::iu2l, 1, length),
+      log2ArrayIndexScale64Bits);
+
+   TR::Node* mask = TR::Node::create(node, TR::lor, 2,
+      TR::Node::create(node, TR::lshl, 2,
+         log2ArrayIndexScale64Bits,
+         TR::Node::lconst(node, 1)),
+      TR::Node::lconst(node, 3));
+
+   TR::Node* lengthToCompare = TR::Node::create(node, TR::land, 2,
+      lengthInBytes,
+      TR::Node::create(node, TR::lxor, 2, mask, TR::Node::lconst(node, -1)));
+
+   TR::Node* mismatchIndex = TR::Node::create(node, TR::arraycmplen, 3);
+   // TODO: replace the following aladd's with generateDataAddrLoadTrees when off-heap memory changes come in
+   mismatchIndex->setAndIncChild(0, TR::Node::create(node, TR::aladd, 2, a, aOffset));
+   mismatchIndex->setAndIncChild(1, TR::Node::create(node, TR::aladd, 2, b, bOffset));
+   mismatchIndex->setAndIncChild(2, lengthToCompare);
+   mismatchIndex->setSymbolReference(getSymRefTab()->findOrCreateArrayCmpLenSymbol());
+
+   TR::Node* thenNode = TR::Node::create(node, TR::ixor, 2,
```

Comment:
Children nodes of ternary select node could have better name then 'elseNode' and 'thenNode'.